### ISSUE TITLE
factorialize: I believe my answer is correct but  cant proceed.

### ISSUE THREAD

**Issue by trextozyne** [freeCodeCamp/freeCodeCamp#14419]
Challenge [Factorialize a Number](https://www.freecodecamp.com/challenges/factorialize-a-number#?solution=var%20globArray%20%3D%20%5B%5D%3B%0A%0Afunction%20factorialize(num)%20%7B%20%0A%20%20var%20i%20%3D%201%3B%20%20var%20j%20%3D%20num%3B%0A%20%20while(i%20%3C%3D%20j)%7B%0A%20%20%20%20globArray.push(i)%3B%0A%20%20%20%20i%2B%2B%3B%0A%20%20%7D%0A%20%20var%20prod%20%3D%20globArray.reduce(function(prev%2C%20next)%7B%20%0A%20%20%20%20return%20prev%20*%20next%3B%0A%20%20%7D%2C%201)%3B%0A%20%20return%20prod%3B%0A%7D%0A%0Afactorialize(20)%3B) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 6.2; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```javascript
var globArray = [];

function factorialize(num) { 
  var i = 1;  var j = num;
  while(i <= j){
    globArray.push(i);
    i++;
  }
  var prod = globArray.reduce(function(prev, next){ 
    return prev * next;
  }, 1);
  return prod;
}

factorialize(20);
```

**Comment by OFFLlNE** [comment]
Thanks @trextozyne for reporting your issue

This is when you use a global variable.

Programs that have tests only run the function itself. And every time the test is ran, the globArray has new values pushed inside of it. You would need to clear the array before running the function each time.

TL;DR: Challenges with multiple tests do not like global variables. For you there is currently an easy fix.. throw that array inside the function

**Comment by trextozyne** [comment]
Wow, that saved me a lot of headache, thanks  it actually worked.
